src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/TrackerObjectManager.NativeAot.cs

```diff
@@ -0,0 +1,543 @@
+        }
+
+        // End the reference tracking process for external object.
+        public static void EndReferenceTracking()
+        {
+            if (!s_HasTrackingStarted || !ShouldWalkExternalObjects())
+            {
+                return;
+            }
+
+            // Let the tracker runtime know the external object walk is done and they need to:
+            // 1. Unpeg all managed object wrappers (mow) if the (mow) needs to be unpegged
+            //       (i.e. when the (mow) is only reachable by other external tracker objects).
+            // 2. Peg all mows if the mow needs to be pegged (i.e. when the above condition is not true)
+            // 3. Unlock reference cache when they are done.
+            Debug.Assert(s_TrackerManager != IntPtr.Zero);
+            IReferenceTrackerManager.ReferenceTrackingCompleted(s_TrackerManager);
+
+            s_IsGlobalPeggingOn = true;
+            s_HasTrackingStarted = false;
+        }
+
+        public static unsafe void RegisterGCCallbacks()
+        {
+            delegate* unmanaged<int, void> gcStartCallback = &GCStartCollection;
+            delegate* unmanaged<int, void> gcStopCallback = &GCStopCollection;
+            delegate* unmanaged<int, void> gcAfterMarkCallback = &GCAfterMarkPhase;
+
+            if (!RuntimeImports.RhRegisterGcCallout(RuntimeImports.GcRestrictedCalloutKind.StartCollection, (IntPtr)gcStartCallback) ||
+                !RuntimeImports.RhRegisterGcCallout(RuntimeImports.GcRestrictedCalloutKind.EndCollection, (IntPtr)gcStopCallback) ||
+                !RuntimeImports.RhRegisterGcCallout(RuntimeImports.GcRestrictedCalloutKind.AfterMarkPhase, (IntPtr)gcAfterMarkCallback))
+            {
+                throw new OutOfMemoryException();
+            }
+        }
+
+        public static bool AddReferencePath(object target, object foundReference)
+        {
+            return s_ReferenceCache.AddDependentHandle(target, foundReference);
+        }
+
+        [UnmanagedCallersOnly]
+        private static void GCStartCollection(int condemnedGeneration)
+        {
+            if (condemnedGeneration >= 2)
+            {
+                s_ReferenceCache.Reset();
+
+                BeginReferenceTracking();
+            }
+        }
+
+        [UnmanagedCallersOnly]
+        private static void GCStopCollection(int condemnedGeneration)
+        {
+            if (condemnedGeneration >= 2)
+            {
+                EndReferenceTracking();
+            }
+        }
+
+        [UnmanagedCallersOnly]
+        private static void GCAfterMarkPhase(int condemnedGeneration)
+        {
+            DetachNonPromotedObjects();
+        }
+
+    }
+
+    // Wrapper for IReferenceTrackerManager
+    internal static unsafe class IReferenceTrackerManager
+    {
+        public static void ReferenceTrackingStarted(IntPtr pThis)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, int>**)pThis)[3](pThis));
```

Comment:
`StdCall` is in some places, but not in others. Delete it from everywhere since it is the default?